tests/core/gen_helpers/gen_from_tests/gen.py

```diff
@@ -1,11 +1,17 @@
+from importlib import reload, import_module
 from inspect import getmembers, isfunction
-from typing import Any, Iterable
+from typing import Any, Callable, Dict, Iterable
 
-from gen_base.gen_typing import TestCase
+from eth2spec.config import config_util
+from eth2spec.utils import bls
+
+from eth2spec.test.context import ALL_CONFIGS, TESTGEN_FORKS, SpecForkName, ConfigName
```

Comment:
Check with @protolambda: I remember you wanted `gen_helpers` to be isolated from `eth2spec`. But here I borrow the `eth2spec` helpers again. 😅 

`eth2spec` context was imported in `gen_helpers/gen_base/gen_runner.py` for skipping the specific tests:
https://github.com/ethereum/eth2.0-specs/blob/c7d975981c04626b72cdc3566914af0aa552b7bb/tests/core/gen_helpers/gen_base/gen_runner.py#L13-L14

This refactoring is for making `main.py` of the state test generators only need to set the handlers.

I want to check if you have a preference against this apporach. 🙂

Reply:
The pytest related test generators all import this package, and importing pyspec makes dependency management even more complicated. I'm afraid of situations where you run a generator, and outputs are incorrect due to a bad build order, or where a pyspec change breaks the generator more quickly. If we really need pyspec here, maybe it's better to change the generator to be a module of eth2spec package, and simplify the dependency management.

Reply:
@protolambda thanks for the response!

https://github.com/ethereum/eth2.0-specs/pull/2199/commits/9cc8567d682ee831adf4a019de31fd0ebbdd30bb is a different approach (different from our chat):
- I moved `gen_helpers` into a module of the `eth2spec` package + some linter fixes.
- The dependencies of `generators/*` modules are only (i) the whole `eth2spec` package (ii) installing `pytest` since it's not part of the default `eth2spec` installation.
- I think now we can keep the generators as status-quo. (independent `venv`s).

Does it look good for you now?